src/wasi.rs

```diff
@@ -0,0 +1,792 @@
+        pub d_ino: __wasi_inode_t,
+        pub d_namlen: u32,
+        pub d_type: __wasi_filetype_t,
+    }
+
+    pub struct __wasi_event_u_fd_readwrite_t {
+        pub nbytes: __wasi_filesize_t,
+        pub flags: __wasi_eventrwflags_t,
+    }
+
+    pub struct __wasi_fdstat_t {
+        pub fs_filetype: __wasi_filetype_t,
+        pub fs_flags: __wasi_fdflags_t,
+        pub fs_rights_base: __wasi_rights_t,
+        pub fs_rights_inheriting: __wasi_rights_t,
+    }
+
+    pub struct __wasi_filestat_t {
+        pub st_dev: __wasi_device_t,
+        pub st_ino: __wasi_inode_t,
+        pub st_filetype: __wasi_filetype_t,
+        pub st_nlink: __wasi_linkcount_t,
+        pub st_size: __wasi_filesize_t,
+        pub st_atim: __wasi_timestamp_t,
+        pub st_mtim: __wasi_timestamp_t,
+        pub st_ctim: __wasi_timestamp_t,
+    }
+
+    pub struct __wasi_ciovec_t {
+        pub buf: *const ::c_void,
+        pub buf_len: usize,
+    }
+
+    pub struct __wasi_iovec_t {
+        pub buf: *mut ::c_void,
+        pub buf_len: usize,
+    }
+
+    pub struct __wasi_subscription_u_clock_t {
+        pub identifier: __wasi_userdata_t,
+        pub clock_id: __wasi_clockid_t,
+        pub timeout: __wasi_timestamp_t,
+        pub precision: __wasi_timestamp_t,
+        pub flags: __wasi_subclockflags_t,
+    }
+
+    pub struct __wasi_subscription_u_fd_readwrite_t {
+        pub fd: __wasi_fd_t,
+    }
+
+    pub struct __wasi_prestat_u_dir_t {
+        pub pr_name_len: usize,
+    }
+}
+
+s_no_extra_traits! {
+    #[allow(missing_debug_implementations)]
+    pub struct __wasi_subscription_t {
+        pub userdata: __wasi_userdata_t,
+        pub type_: __wasi_eventtype_t,
+        pub u: __wasi_subscription_u,
+    }
+
+    #[allow(missing_debug_implementations)]
+    pub struct __wasi_event_t {
+        pub userdata: __wasi_userdata_t,
+        pub error: __wasi_errno_t,
+        pub type_: __wasi_eventtype_t,
+        pub u: __wasi_event_u,
+    }
+
+    #[allow(missing_debug_implementations)]
+    pub union __wasi_event_u {
+        pub fd_readwrite: __wasi_event_u_fd_readwrite_t,
+        _bindgen_union_align: [u64; 2usize],
+    }
+
+    #[allow(missing_debug_implementations)]
+    pub union __wasi_subscription_u {
+        pub clock: __wasi_subscription_u_clock_t,
+        pub fd_readwrite:
+            __wasi_subscription_u_fd_readwrite_t,
+        _bindgen_union_align: [u64; 5usize],
+    }
+
+    #[allow(missing_debug_implementations)]
+    pub struct __wasi_prestat_t {
+        pub pr_type: __wasi_preopentype_t,
+        pub u: __wasi_prestat_u,
+    }
+
+    #[allow(missing_debug_implementations)]
+    pub union __wasi_prestat_u {
+        pub dir: __wasi_prestat_u_dir_t,
+    }
+
+}
+
+pub static CLOCK_REALTIME: clockid_t = unsafe { &__CLOCK_REALTIME_ADDR };
+pub static CLOCK_MONOTONIC: clockid_t = unsafe { &__CLOCK_MONOTONIC_ADDR };
+pub static CLOCK_MONOTONIC_RAW: clockid_t =
+    unsafe { &__CLOCK_MONOTONIC_RAW_ADDR };
```

Comment:
So how are these expressend in C? 

I see that the `__CLOCK_..._ADDR` are zero-sized statics, and that this takes a `&'static` reference to them. Are they macros in C?